pkg/envoy/rds/route/route_config.go

```diff
@@ -192,8 +192,12 @@ func buildInboundRoutes(rules []*trafficpolicy.Rule) []*xds_route.Route {
 func buildOutboundRoutes(outRoutes []*trafficpolicy.RouteWeightedClusters) []*xds_route.Route {
 	var routes []*xds_route.Route
 	for _, outRoute := range outRoutes {
-		emptyHeaders := map[string]string{}
-		routes = append(routes, buildRoute(trafficpolicy.PathMatchRegex, constants.RegexMatchAll, constants.WildcardHTTPMethod, emptyHeaders, outRoute.WeightedClusters, outRoute.RetryPolicy))
+		// Create temp variable to avoid potentially overwriting the loop variable
+		tempOutbound := outRoute
```

Comment:
Since `outRoute` is a pointer:
```suggestion
		tempOutbound := *outRoute
```
Otherwise both will update the same underlying memory.